### PR TITLE
Allow non-String tag values

### PR DESCRIPTION
**Pull request by estolfo**
This is a breaking change - tag values were previously always converted to a string. This PR allows non-String tag values, while still truncating String values.

### CODE REVIEW

File: lib/elastic_apm/instrumenter.rb [hunk 1]
```diff
@@ -195,7 +195,7 @@ def set_tag(key, value)
       return unless current_transaction
 
       key = key.to_s.gsub(/[\."\*]/, '_').to_sym
```
Comment (mikker) [inline lib/elastic_apm/instrumenter.rb]: We could save a string allocation by using `gsub!` instead. Could be too insignificant to worry about. How does `gsub!` even work with `frozen_string_literal`? Questions, questions.

**Comment by mikker** [comment]
Approved, still needs
- [ ] docs
- [ ] changelog entry

**Comment by estolfo** [comment]
@mikker what docs are needed?

File: lib/elastic_apm/instrumenter.rb (hunk 1 above)
Comment (estolfo) [reply lib/elastic_apm/instrumenter.rb]: You can't call `gsub!` on a frozen string, unfortunately. I can't think of another way to do this without creating another String. Let me know if you have any ideas...

Comment (estolfo) [reply lib/elastic_apm/instrumenter.rb]: We could do it if we wrote this in a c extension ; )

File: lib/elastic_apm.rb [hunk 2]
```diff
@@ -361,7 +361,7 @@ def report_message(message, context: nil, **attrs)
     # Set a _tag_ value for the current transaction
     #
     # @param key [String,Symbol] A key
-    # @param value [Object] A value (will be converted to string)
```
Comment (axw) [inline lib/elastic_apm.rb]: should we perhaps leave `set_tag` alone, and make this change on `set_label` in #543? `set_tag` would call `set_label(key, value.to_s)`?

Comment (mikker) [reply lib/elastic_apm.rb]: I like this idea.

**Comment by mikker** [comment]
@estolfo `docs/api.asciidoc` is one. Perhaps do project-wide search, see if we mention `set_tag` anywhere else.

File: lib/elastic_apm.rb (hunk 2 above)
Comment (estolfo) [reply lib/elastic_apm.rb]: Yes, this is great! That way, it won't be a breaking change to allow boolean and numerical tag values. Thanks for the suggestion @axw

**Comment by estolfo** [comment]
Changes have been added to PR #543

**Pull request closed without merging**